src/components/StatisticsCard/Battles.js

```diff
@@ -3,10 +3,11 @@ import { OverlayTrigger, Tooltip as Tips } from 'react-bootstrap';
 import BarChart from '../Charts/BarChart';
 
 const Battles = (props) => {
-  const lastSnapshot = props.tankStats.data.snapshots.at(-1);
+  const stats = props.tankStatsCard ? props.tankStatsCard : props.accountStats?.data;
+  const lastSnapshot = stats.snapshots.at(-1);
   const dataForTables = props.filteredStats.dataForTables;
```

Comment:
You can use here destructuring assignment
```suggestion
  const { dataForTables } = props.filteredStats;
```